### ISSUE TITLE
[significant_drop_tightening] Functions that return literals should not trigger the lint

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#11160]
Literals are cheap expressions that won't interfere with the release of a lock.

```rust
pub fn foo() -> bool {
    let mutex = Mutex::new(1i32);
    let lock = mutex.lock().unwrap();
    let _ = lock.abs();
    true
}
```

cc https://github.com/rust-lang/rust-clippy/issues/11125#issuecomment-1635705282
